collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -575,6 +571,11 @@ func (r *receiverStream) recvOne(streamCtx context.Context, serverStream anyStre
 		return status.Errorf(codes.Internal, "arrow metadata error: %v", err)
 	}
 
+	// start this span after hrcv.combineHeaders returns extracted context. This will allow this span
+	// to be a part of the data path trace instead of only being included as a child of the stream inflight trace.
+	inflightCtx, span := r.tracer.Start(inflightCtx, "otel_arrow_stream_recv")
+	defer span.End()
+
```

Comment:
Looks good. This means the Authentication and semaphore Acquire will be part of the data-path context.